docs/rules/unicode-bom.md

```diff
@@ -0,0 +1,60 @@
+# Allow or disallow the Unicode Byte Order Mark (BOM) (unicode-bom)
+
+The Unicode Byte Order Mark (BOM) is used to specify whether code units are big
+endian or little endian. That is, whether the most significant or least
+significant bytes come first. UTF-8 does not require a BOM because byte ordering
+does not matter when characters are a single byte. Since UTF-8 is the dominant
+encoding of the web, we make `"never"` the default option.
+
+## Rule Details
+
+If the `"never"` option is used, this rule requires that the file does not begin
+with the unicode BOM character U+FEFF.
+
+## Options
+
+This rule has a string option:
+
+* `"always"` the unicode BOM is always allowed
```

Comment:
Yes, that would be how I think the rule should be enforced. Users can
simply disable the rule to achieve "either/or". See the `semi` rule for
another example.
On May 19, 2016 5:18 PM, "Andrew Johnston" <notifications@github.com> wrote:

> In docs/rules/unicode-bom.md
> <https://github.com/eslint/eslint/pull/6230#discussion_r63960084>:
>
> > +The Unicode Byte Order Mark (BOM) is used to specify whether code units are big
> > +endian or little endian. That is, whether the most significant or least
> > +significant bytes come first. UTF-8 does not require a BOM because byte ordering
> > +does not matter when characters are a single byte. Since UTF-8 is the dominant
> > +encoding of the web, we make `"never"` the default option.
> > +
> > +## Rule Details
> > +
> > +If the `"never"` option is used, this rule requires that the file does not begin
> > +with the unicode BOM character U+FEFF.
> > +
> > +## Options
> > +
> > +This rule has a string option:
> > +
> > +* `"always"` the unicode BOM is always allowed
>
> I may have misunderstood #5502
> <https://github.com/eslint/eslint/issues/5502>. What is the correct
> behaviour? If "always" then the file must have the BOM and if "never" then
> it must not have the BOM?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/eslint/eslint/pull/6230/files/a6d4b4a994ee6373aed76b78e6a2567c3ff90c92#r63960084>
>